tests/test_home_page.py

```diff
@@ -1,38 +1,87 @@
 """Test home page."""
 
+import mongoengine
+import mongomock
 import pytest
 from selenium.webdriver.remote.webelement import WebElement
 from tests import helper_test_functions as helper
 from tests import settings
 
+from dashboard.models import user
+from dashboard.pages.index import controller
+
 from .settings import DriverType
 
-CREATE_DASHBOARD_BUTTON_ID = "create-dashboard"
+CREATE_DASHBOARD_BUTTON_ID = "create-dashboard-btn"
 LATEST_OPENED_DASHBOARDS_ID = "latest-opened-dashboards"
 SHARED_DASHBOARDS_ID = "shared-dashboards"
+EMPTY_TITLE_OUTPUT = (True, True, False)
+EMPTY_DESC_OUTPUT = (True, False, True)
+EMPTY_TITLE_DESC_OUTPUT = (True, True, True)
+VALID_OUTPUT = (False, False, False)
+
+
+@pytest.fixture(autouse=True)
+def connection():
+    """Connect mongoengine to mongomock and return client."""
+    conn = mongoengine.connect(
+        db="dashboard",
+        host="mongodb://localhost",
+        mongo_client_class=mongomock.MongoClient,
+        uuidRepresentation="standard",
+    )
+
+    return conn
 
 
-"""
-TODO
-Add tests for modals when modal component is finished
-"""
+@pytest.fixture(autouse=True)
+def example_user():
+    """Example user."""
+    username = "dashboards-page-test-user"
+    usr = user.login_user(username)
+    return usr
 
 
 @pytest.mark.test_home_page
 class TestHomePage:
-    """A class to group functions to test the dashboard capabilities."""
+    """A class to group functions to test home page."""
+
+    pytest.mark.usefixtures("start_server")
 
-    @pytest.mark.usefixtures("start_server")
     def test_create_dashboard_button(self, browser_driver: DriverType) -> None:
-        """Check that the create dashboard button exists.
+        """Test the create dashboard button.
+
+        Args:
+            browser_driver (DriverType): webdriver used for selenium
+        """
+        browser_driver.get(settings.HOME_PAGE_URL)
+
+        create_dashboard: WebElement = helper.get_element_by_id(
+            browser_driver, CREATE_DASHBOARD_BUTTON_ID
+        )
+        create_dashboard.click()
+
+    @pytest.mark.usefixtures("start_server")
+    def test_create_dashboard(self, browser_driver: DriverType) -> None:
+        """Tries to create dashboards using callback.
 
         Args:
             browser_driver (DriverType): webdriver used for selenium
         """
-        browser_driver.get(settings.URL)
+        browser_driver.get(settings.HOME_PAGE_URL)
 
-        element: WebElement = helper.get_element_by_id(browser_driver, CREATE_DASHBOARD_BUTTON_ID)
-        assert element
+        create_dashboard: WebElement = helper.get_element_by_id(
+            browser_driver, CREATE_DASHBOARD_BUTTON_ID
+        )
+        create_dashboard.click()
```

Comment:
Is the opening of the modal important in this step if you are just going to unit test the callback? In my opinion, either just test the callback functions with different parameters, or just test the user interface with different inputs using the webdriver.